### PR TITLE
Use AzureCLI@2 and serviceConnection to run CI/CD tests instead of manually logging in az

### PR DESCRIPTION
**Pull request by vhvb1989**
This PR removes the manual logging to `az` using client secrets.

Instead, it uses the Ado task `AzureCLI@2` and a service connection.  The login is scoped to the task now.

### CODE REVIEW

File: cli/azd/pkg/project/framework_service_docker.go [hunk 1]
```diff
@@ -215,7 +215,7 @@ func (p *dockerProject) Build(
 				}
 				return result
 			}
-			// resolve parameters for build args and secrets
+			// resolve parameters for build args and secrets...
```
Comment (wbreza) [inline cli/azd/pkg/project/framework_service_docker.go:218]: Rollback for this PR.

**Review by wbreza — APPROVED** [review]
LGTM 🚀

**Review by ellismg — APPROVED** [review]
LGTM once you do the rollback that @wbreza mentioned.  Thanks for fixing this!

File: eng/pipelines/templates/jobs/build-cli.yml [hunk 2]
```diff
@@ -89,24 +89,22 @@ jobs:
...
-      - task: PowerShell@2
+      - task: AzureCLI@2
         condition: and(succeeded(), ne(variables['Skip.LiveTest'], 'true'))
         inputs:
-          pwsh: true
-          targetType: filePath
-          filePath: cli/azd/ci-test.ps1
-          arguments: >-
+          azureSubscription: azure-sdk-tests
+          keepAzSessionActive: true
```
Comment (danieljurek) [inline eng/pipelines/templates/jobs/build-cli.yml:100]: Add a comment here that this is an experimental feature and may break

**Review by danieljurek — COMMENTED** [review]
Can `azd-login.yml` also be eliminated here?

**Comment by vhvb1989** [comment]
> Can `azd-login.yml` also be eliminated here?

No. Maybe in the future, once we have an Azdo task to run azd.  For now, azd needs to be manually logged in.

**Pull request merged**